shell/platform/darwin/macos/framework/Source/FlutterTextInputSemanticsObject.mm

```diff
@@ -0,0 +1,205 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterTextInputSemanticsObject.h"
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterEngine_Internal.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterViewController_Internal.h"
+
+#include "flutter/third_party/accessibility/ax/ax_action_data.h"
+#include "flutter/third_party/accessibility/gfx/geometry/rect_conversions.h"
+#include "flutter/third_party/accessibility/gfx/mac/coordinate_conversion.h"
+
+#pragma mark - FlutterTextFieldCell declaration
+/**
+ * A convenient class that lets you to set a custom field editor for a
+ * NSTextField.
+ *
+ * The FlutterTextField uses this class set the FlutterTextInputPlugin as
+ * its field editor.
+ */
+@interface FlutterTextFieldCell : NSTextFieldCell
+
+/**
+ * Initializes the NSCell for the input NSTextField that will use the input
+ * NSTextView as its field editor.
+ */
+- (instancetype)initWithTextField:(NSTextField*)textField fieldEditor:(NSTextView*)editor;
+
+@end
+
+#pragma mark - FlutterTextFieldCell implementation
+
+@implementation FlutterTextFieldCell {
+  // NSTextView does not support _weak reference, so we have to
+  // use __unsafe_unretained and manage the reference ourselves.
```

Comment:
I'm not sure what this comment means in practice. Who is "ourselves"? Because there's no API for clearing the editor, which means that an instance of this class cannot possibly be correctly managing it.